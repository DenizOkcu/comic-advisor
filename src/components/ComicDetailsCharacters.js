import React from "react";

const ComicDetailsCharacters = ({ characters }) => {
  // ids fÃ¼r den key im serializer vergeben
  const characterItems = characters.map(character => (
    <li key={character.id}>{`${character.name}`}</li>
  ));

  return (
    <section>
      <b>Characters in this issue:</b>
      <ul>{characterItems}</ul>
    </section>
  );
};

export default ComicDetailsCharacters;
