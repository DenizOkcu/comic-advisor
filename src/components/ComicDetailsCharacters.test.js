import React from "react";
import { shallow } from "enzyme";
import ComicDetailsCharacters from "./ComicDetailsCharacters";

describe("ComicDetailsCharacters", () => {
  it('should render correctly in "debug" mode', () => {
    const characters = [{ id: 0, name: "IronMan" }, { id: 1, name: "Hulk" }];
    const component = shallow(
      <ComicDetailsCharacters characters={characters} debug />
    );

    expect(component).toMatchSnapshot();
  });
});
