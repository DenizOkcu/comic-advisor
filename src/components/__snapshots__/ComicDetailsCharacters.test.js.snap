// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComicDetailsCharacters should render correctly in "debug" mode 1`] = `
<section>
  <b>
    Characters in this issue:
  </b>
  <ul>
    <li
      key="0"
    >
      IronMan
    </li>
    <li
      key="1"
    >
      Hulk
    </li>
  </ul>
</section>
`;
